### PR TITLE
Rfc85 dynamic virtual study

### PR DESCRIPTION
**Pull request by forus**
Fix # (see https://help.github.com/en/articles/closing-issues-using-keywords)

Describe changes proposed in this pull request:
- add a flag to indicate if virtual study dynamic
- calculate sample IDs using virtual study view filters if study is dynamic

# Checks
- [ ] The commit log is comprehensible. It follows [7 rules of great commit messages](http://chris.beams.io/posts/git-commit/). We can fix this during merge by using a squash+merge if necessary
- [ ] Has tests or has a separate issue that describes the types of test that should be created. If no test is included it should explicitly be mentioned in the PR why there is no test.
- [ ] Is this PR adding logic based on one or more **clinical** attributes? If yes, please make sure validation for this attribute is also present in the data validation / data loading layers (in backend repo) and documented in [File-Formats Clinical data section](https://github.com/cBioPortal/cbioportal/blob/master/docs/File-Formats.md#clinical-data)!
- [ ] Make sure your PR has one of the labels defined in https://github.com/cBioPortal/cbioportal/blob/master/.github/release-drafter.yml

# Any screenshots or GIFs?
If this is a new visual feature please add a before/after screenshot or gif
here with e.g. [Giphy CAPTURE](https://giphy.com/apps/giphycapture) or [Peek](https://github.com/phw/peek)

# Notify reviewers
Read our [Pull request merging
policy](../CONTRIBUTING.md#pull-request-merging-policy). It can help to figure out who worked on the
file before you. Please use `git blame <filename>` to determine that
and notify them either through slack or by assigning them as a reviewer on the PR

### CODE REVIEW

File: src/main/java/org/cbioportal/web/SessionServiceController.java [hunk 1]
```diff
@@ -232,6 +244,29 @@ public ResponseEntity<Session> getSession(@PathVariable Session.SessionType type
...
+                Collectors.groupingBy(
+                    SampleIdentifier::getStudyId,
+                    Collectors.mapping(SampleIdentifier::getSampleId, Collectors.toSet())));
+        Set<VirtualStudySamples> virtualStudySamples = sampleIdsByStudyId.entrySet().stream().map(entry -> {
+            VirtualStudySamples vss = new VirtualStudySamples();
+            vss.setId(entry.getKey());
+            vss.setSamples(entry.getValue());
+            return vss;
+        }).collect(Collectors.toSet());
+        virtualStudyData.setStudies(virtualStudySamples);
```
Comment (pieterlukasse) [inline src/main/java/org/cbioportal/web/SessionServiceController.java:267]: maybe a separate ticket: should `setStudies` be renamed to `setSamples`?

File: src/main/java/org/cbioportal/web/SessionServiceController.java [hunk 2]
```diff
@@ -232,6 +244,29 @@ public ResponseEntity<Session> getSession(@PathVariable Session.SessionType type
         }
     }
 
+    /**
+     * This method populates the `virtualStudyData` object with a new set of sample IDs retrieved as the result of executing a query based on virtual study view filters.
+     * It first applies the filters defined within the study view, runs the query to fetch the relevant sample IDs, and then updates the virtualStudyData to reflect these fresh results.
+     * This ensures that the virtual study contains the latest sample IDs.
+     * @param virtualStudyData
+     */
```
Comment (pieterlukasse) [inline src/main/java/org/cbioportal/web/SessionServiceController.java:252]: Nice! Quick question: will this be cached? If yes, how?

**Review by pieterlukasse — COMMENTED** [review]
code looks good to me. Thanks @forus !

I added two questions.

File: src/main/java/org/cbioportal/web/SessionServiceController.java [hunk 3]
```diff
@@ -232,6 +244,29 @@ public ResponseEntity<Session> getSession(@PathVariable Session.SessionType type
...
+            .collect(
+                Collectors.groupingBy(
+                    SampleIdentifier::getStudyId,
+                    Collectors.mapping(SampleIdentifier::getSampleId, Collectors.toSet())));
+        Set<VirtualStudySamples> virtualStudySamples = sampleIdsByStudyId.entrySet().stream().map(entry -> {
+            VirtualStudySamples vss = new VirtualStudySamples();
+            vss.setId(entry.getKey());
+            vss.setSamples(entry.getValue());
+            return vss;
+        }).collect(Collectors.toSet());
```
Comment (haynescd) [inline src/main/java/org/cbioportal/web/SessionServiceController.java:266]: Opinion, but I would extract this into a function called `extractVirtualStudySamples` or something similar.

File: src/main/java/org/cbioportal/web/parameter/VirtualStudyData.java [hunk 4]
```diff
@@ -24,6 +24,8 @@ public class VirtualStudyData implements Serializable {
     private String typeOfCancerId;
     private String pmid;
 
+    private Boolean dynamic;
```
Comment (haynescd) [inline src/main/java/org/cbioportal/web/parameter/VirtualStudyData.java:27]: Is it possible to be NULL? is that why you are choosing not to use the primitive type boolean?

File: src/main/java/org/cbioportal/web/SessionServiceController.java (hunk 2 above)
Comment (forus) [reply src/main/java/org/cbioportal/web/SessionServiceController.java:252]: @pieterlukasse this method is not cached, but `studyViewFilterApplier.apply(virtualStudyData.getStudyViewFilter());` is. I don't think we should cache this method as it'd be cheap transformation.

File: src/main/java/org/cbioportal/web/parameter/VirtualStudyData.java (hunk 4 above)
Comment (forus) [reply src/main/java/org/cbioportal/web/parameter/VirtualStudyData.java:27]: @haynescd VS created before this change will have this field set to null. I am actually not sure what'd happen if this field would be primitive. I'd expect serialiser to complain

File: src/main/java/org/cbioportal/web/SessionServiceController.java (hunk 3 above)
Comment (forus) [reply src/main/java/org/cbioportal/web/SessionServiceController.java]: @haynescd Please have a look https://github.com/cBioPortal/cbioportal/pull/11040/commits/eda7367cf717a30aa1c6a098f407c663015f5757

File: src/main/java/org/cbioportal/web/SessionServiceController.java (hunk 1 above)
Comment (forus) [reply src/main/java/org/cbioportal/web/SessionServiceController.java]: @pieterlukasse https://github.com/cBioPortal/cbioportal/issues/11094

**Review by haynescd — APPROVED** [review]

**Pull request merged**